### DISCUSSION TITLE
Resize, Blending modes and Merge Images and renaming them.

### DISCUSSION THREAD

**Discussion by Archive-d** [ImageMagick/ImageMagick#5144] [Help] [answered]
Operating System: Windows 10

Hello everyone,
I am a complete newbie to this.
I need help in merging multiple images from different folders in a pattern and saving them like the image here.
![Screenshot-(49)](https://user-images.githubusercontent.com/76863644/167780689-dc699aa9-a1b7-4b45-9a20-48b7d0e5fc9a.jpg)

Explaining it further
I want the second image to follow same pattern.
Example: Images in 2 folders
Folder 1: a, b, c...  (100s of images)
Folder 2: 1, 2, 3...

To be saved as a1, a2, a3, b1, b2, b3, c1..... in folder 3

Also if possible,
1, Apply a blending mode like multiply or overlay or screen like in photoshop, and then merge the images, 
2, Resize folder 1 images by percentage before merging them. So, the sword will be like 30% smaller in the final image.

PS. The foreground and background images are of same size i.e, 1080x1080
     The sword is just an example. Actual images will differ.

**Comment by snibgo** [comment]
What version of IM, on what platform? I will assume IM v7 on Windows.

You have hundreds of images, each 1080x1080 pixels. Perhaps all the images can be stored in memory at the same time. If so, then the command could be:

```
magick folder2\*.png null: folder1\*.png -compose Over -layers composite folder3\out_%06d.png
```
On "-layers composite", see https://imagemagick.org/script/command-line-options.php#layers . The blending mode is whatever you want. If you want to resize all the images in folder 1 by the same percentage:

```
magick folder2\*.png null: ( folder1\*.png -resize 30% ) -compose Over -layers composite folder3\out_%06d.png
```

For Windows BAT, double every percent %, so "folder3\out_%%06d.png" etc.

This will not name the output files as you want.

An alternative is to have a shell loop, that loops through one directory and for each file loops through the other directory, then for each file it executes a simple magick command. It would use an environment variable for the input and output files, so the output names can be as you want. This method would be much slower than the first, but it doesn't need to store all the images in memory at the same time.

**Reply by Archive-d** [reply]
Hello,
Thank you so much for the quick reply.

I tried running the script with all my coding knowledge (~0) and it worked. However, it merged in pattern of 
F1 containing 3 images (a b c) F2 containing 6 images (1-6)
a1, b2, c3, 4, 5, 6.  and not a1, a2, a3,....a6, b1, b2, b3...

Could you please help on this. 

I forgot to mention,
Folder 1 has 100s of foreground images and folder 2 has limited background images <10.

How do I do the second method you mentioned? I could use an excel sheet and cmd script I came across on the internet to do the naming.  But would be amazing if it is automated like you mentioned. 

I am fine with storing images on memory if required.

PS. IM 7.1 Q16 x64 on Windows 10.

Thank you again for taking time for this.

**Comment by snibgo** [comment, marked as answer]
Sorry, I think my previous reply was wrong, and doesn't do what you want.

Suppose you have 200 images in folder1, and 10 images in folder2. Then you want to combine every image from folder1 with every image in folder2, creating 2000 outputs. Is that correct?

Then "-layers composite" won't do that. It will give only 10 outputs that are combinations of two inputs. So forget what I said about that.

The best way would be the "alternative" I gave. I assume folder1 and folder2 contain only image files. A Windows BAT script could be:
```
for %%A in (folder1\*) do (
  for %%B in (folder2\*) do (
    magick %%B %%A -compose Over -composite outdir\%%~nA%%~nB.png
  )
)
```
I haven't tested the script, and may have a detail wrong.

Each output will be named using a combination of the names in folder1 and folder2. I assumed you want PNG outputs. Perhaps you want JPG or something else.

**Comment by Archive-d** [comment]
Oh, This worked.
I can't thank you enough for this. *insert crying emoji*
Hope you have an amazing day.
